homeassistant/components/openweathermap/const.py

```diff
@@ -234,7 +242,10 @@
 }
 FORECAST_SENSOR_TYPES = {
     ATTR_FORECAST_CONDITION: {SENSOR_NAME: "Condition"},
-    ATTR_FORECAST_PRECIPITATION: {SENSOR_NAME: "Precipitation"},
+    ATTR_FORECAST_PRECIPITATION: {
+        SENSOR_NAME: "Precipitation",
+        SENSOR_UNIT: PRECIPITATION_MILLIMETERS_PER_HOUR,
+    },
```

Comment:
Should the SENSOR_UNIT for ATTR_FORECAST_PRECIPITATION just be "mm" instead of "mm/h"?

If the forecast mode is "onecall_daily" or "daily", I think (from what I can tell from the [docs](https://openweathermap.org/api/one-call-api#parameter) and testing different latitude/longitude) the OpenWeatherMap api is responding with the total forecasted rain/snow for the entire day (mm) instead of a precipitation intensity (mm/h).

Similarly, if the forecast mode is "onecall_hourly" or "hourly", the api responds with the total forecasted rain/snow for the hour. In this case, "mm/h" works but I think "mm" also applies since the forecast period is one hour.

Reply:
I'm still coming to grips with this integration, and I don't have a paid subscription for OWM so I'm only looking at their One Call API. The forecast "mode" seems to be used to decide whether the data is pulled out of the `hourly` or `daily` keys in the JSON, but it's also pulling some data out of the `current` key. There's also a `minutely` key (which also includes precipitation data) that doesn't seem to be used.

Under the `current` and `hourly` keys, precipitation for rain and snow is provided as _"volume for last hour, mm"_ so I viewed this as precipitation intensity with "mm/h" units. You make a good point about the forecast period being one hour - that means a "mm" unit also makes sense. I'm chasing "feature parity" between this integration and what I had before with the DarkSky integration (which provided precipitation intensity) so I chose the "mm/h" route.

Under the `daily` key, precipitation for rain and snow is provided as _"volume, mm"_ - you're right, this would make more sense having a unit of "mm".

The right thing<sup>TM</sup> to do seems to be to use a sensor with "mm" units if the forecast is "onecall_daily" or "daily" and use a sensor with "mm/h" if the forecast is "onecall_hourly" or "hourly". Maybe even better, instead of different units, use different names for the sensors that better explain the time period over which the volume data is being recorded over? This might apply to some of the other sensors as well, so it might warrant a more comprehensive PR that also makes sure the sensors are pulling data consistently from either the `hourly` or `daily` keys, depending on the forecast mode, and not from `current`.